src/api.jl

```diff
@@ -1,55 +1,224 @@
+    if type1 == type2
+        return set_distance(po1.points, po2.points, distance)
+    else
+        throw(ArgumentError("Both periodic orbits have to be either discrete or continuous."))
+    end
+end
+
+
+"""
+    equal(po1::PeriodicOrbit, po2::PeriodicOrbit; 
+    Tthres=1e-3, dthres=1e-3, dist=StrictlyMinimumDistance(true, Euclidean())) → true/false
+
+Returns `true` if the periodic orbits `po1` and `po2` are equal within the given thresholds.
+Distance between the orbits is computed using the given distance function `dist`.
+"""
+function equal( # better name maybe? isapprox?
+        po1::PeriodicOrbit, po2::PeriodicOrbit;
+        Tthres = 1e-3,
+        dthres = 1e-3,
+        dist = StrictlyMinimumDistance(true, Euclidean())
+    )
+    if abs(po1.T - po2.T) > Tthres
+        return false
+    end
+    d = distance(po1, po2, dist)
+    return d < dthres
+end
+
+"""
+    true_period(ds::DynamicalSystem, po::PeriodicOrbit, atol=1e-4) → po
 
-@kwdef struct Algorithm1 <: PeriodicOrbitFinder
-    param1 = 1
-    param2 = 2
-    param3 = 3
+Computes the true (minimal) period of the periodic orbit `po` of the dynamical system `ds`.
+Returns the periodic orbit with the true period.
+"""
+function true_period(ds::DynamicalSystem, po::PeriodicOrbit, atol=1e-4)
+    type1 = isdiscretetime(ds)
+    type2 = isdiscretetime(po)
+    if type1 == type2
+        return _true_period(ds, po, atol)
+    else
+        throw(ArgumentError("Both the periodic orbit and the dynamical system have to be either discrete or continuous."))
+    end
 end
 
-function periodic_orbit(ds::DynamicalSystem, alg::PeriodicOrbitFinder, ig::InitialGuess = InitialGuess(ds))
+function _true_period(ds::DiscreteTimeDynamicalSystem, po::PeriodicOrbit, atol)
+    u = po.points[1]
+    for n in 1:po.T-1
+        po.T % n != 0 && continue
+        reinit!(ds, u)
+        step!(ds, n)
+        if norm(u - current_state(ds)) < atol
+            return PeriodicOrbit(ds, u, n, 1)
+        end
+    end
+    return po
+end
+
+function _true_period(ds::ContinuousTimeDynamicalSystem, po::PeriodicOrbit, atol)
+    # if we encounter an algorithm that would return PO with non-true period, we will implement this function
+    return po.T
+end
+
+
+"""
+    uniquepos(pos::Vector{PeriodicOrbit}, atol=1e-6) → Vector{PeriodicOrbit}
+
+Returns a vector of unique periodic orbits from the vector `pos` of periodic orbits.
+By unique we mean that the distance between any two periodic orbits in the vector is 
+greater than `atol`. To see details about the distance function, see `distance`.
+"""
+function uniquepos(pos::Vector{PeriodicOrbit{D, B, R}}, atol::Real=1e-6) where {D, B, R}
+    unique_pos = PeriodicOrbit[]
+    
+    filter(po -> begin
+        if all(unique_po -> distance(unique_po, po) > atol, unique_pos)
+            push!(unique_pos, po)
+            true  # Keep this element
+        else
+            false  # Exclude this element
+        end
+    end, pos)
```

Comment:
Yes this is bad code. I still try to use filter in the new version but maybe it would be easier to just have standard two for loops and comparing each element with each element. In the new version I use `push!` to store unique POs and `filter!` to remove duplicates from the original container.